hazelcast-client/src/main/java/com/hazelcast/client/config/XmlClientConfigBuilder.java

```diff
@@ -129,7 +129,7 @@ protected Document parse(InputStream inputStream) throws Exception {
                     + lineSeparator + "Exception: " + e.getMessage()
                     + lineSeparator + "HazelcastClient startup interrupted.";
             LOGGER.severe(msg);
-            throw e;
+            throw new InvalidConfigurationException(e.getMessage());
```

Comment:
This could be problematic because you are throwing the original cause. This makes configuring out the exact cause a problem. In most cases you want to do 'throw new InvalidConfigurationException(e.getMessage(), e);